src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/steps/base-step/base-step.component.ts

```diff
@@ -27,30 +36,51 @@ export class BaseStepComponent extends BaseComponent implements OnInit {
     @Input() public sectionStepRadioData: RadioControlType[];
     @Input() public sectionFormData: JsonFormData;
     @Input() public defaultKind: string;
-    @Input() public groupName: string;
+    @Input() public groupName: SetPollingSourceSection;
     @Input() public title: string;
+    @Input() public eventHistory: DatasetHistoryUpdate;
+    @Input() public eventYamlByHash: string;
+    private editFormValue: EditFormType;
     public controlType: typeof ControlType = ControlType;
     public readonly KIND_NAME_CONTROL = "kind";
     public readonly SCHEMA_NAME_CONTROL = "schema";
-    private readonly DEFAULT_EVENT_TIME_SOURCE = "fromMetadata";
+    private readonly DEFAULT_EVENT_TIME_SOURCE =
+        EventTimeSourceKind.FROM_METADATA;
 
     constructor(
         private rootFormGroupDirective: FormGroupDirective,
         private fb: FormBuilder,
+        private cdr: ChangeDetectorRef,
+        private editService: EditPollingSourceService,
     ) {
         super();
     }
 
+    public get sectionForm(): FormGroup {
+        return this.parentForm.get(this.groupName) as FormGroup;
+    }
+
     ngOnInit(): void {
         this.parentForm = this.rootFormGroupDirective.form;
         this.initForm(
             this.sectionForm.get(this.KIND_NAME_CONTROL)?.value as string,
         );
         this.chooseFetchKind();
+        this.initEditForm();
+        this.cdr.detectChanges();
     }
 
-    public get sectionForm(): FormGroup {
-        return this.parentForm.get(this.groupName) as FormGroup;
+    private initEditForm(): void {
+        if (this.eventHistory.history.length !== 1) {
+            this.editFormValue = this.editService.parseEventFromYaml(
+                this.eventYamlByHash,
+            );
+            this.editService.patchFormValues(
+                this.sectionForm,
+                this.editFormValue,
+                this.groupName,
+            );
+        }
```

Comment:
Looking at this code, it seems we do not need to pass the entire history here at all. It is enough to pass a nullable `eventYamlByHash` to the step components, and if it is not null, then patch the form. History is only used here.

Reply:
Fixed